### ISSUE TITLE
Styling not applied when passing a Component as props.

### ISSUE THREAD

**Issue by santaclauze** [styled-components/styled-components#377]
## Version
1.1.2

I have created a Component as follows: `<HeaderNavBarSlide  button={this.props.button} />` 
It is made via `<HeaderNavBar button={this.props.button} />` and wrapped with a Hoc `composeSlide`.

The props button allows for a Component Button to be passed and if not, will return a value by default: 
This is my base component:
```
class HeaderNavBar extends React.Component {
    static propTypes = {
        className: PropTypes.string,
        button: PropTypes.bool
    }
    constructor(props) {
       super(props);
       const isButton = this.props.button;
       if (!isButton) {
          this.button = Button;
       } else {
          this.button = this.props.button;
       }
      render() {
        const ButtonToggle = this.button;
     }
     return (
        <Header className={this.props.className}>
              <ButtonToggle
                    className={'navbar-toggle}
                    type="button"
                    onClick={this.props.composePush.onClick}
               />
     )
```

`Button` is a Component imported and will be the value by default.

I then apply to my new Component :
`import Button1 from 'components/Button1';`
`<HeaderNavBarSlide  button={Button1}` 

## Expected Behavior
Have the css attached to Button1 to be displayed.

## Actual Behavior
See's the Button1 Component passed via the props.button, also identifies the vendor prefix generated by styled for Button1 BUT does not apply the css.

![image](https://cloud.githubusercontent.com/assets/17665998/21955192/404255b8-da98-11e6-8a54-c360b379945e.png)

Also, when checking the css rules attached to the vendor prefix via styled css, it is also attached:

![image](https://cloud.githubusercontent.com/assets/17665998/21955163/e71fd9c4-da97-11e6-8122-4482e3bd76aa.png)

However the css is not displayed in the web browser page nor is it visible in the console styles:
![image](https://cloud.githubusercontent.com/assets/17665998/21955196/66b4302c-da98-11e6-89b8-d7ee3045b939.png)

Is this a bug or am I doing something wrong?

**Comment by kopax** [comment]
I would say that if styled as generated it's vendor class, it's content and appended to the page the rendered css, this means the styled-component as been applied.

What happen if you remove the whole css and just leave the `.fvLAUo` class, does it apply? 
If yes, you probably have a css syntax. 

This should be corrected in styled-components 2.0 using a babel plugin that would minify and check the css syntax. (According to https://github.com/styled-components/babel-plugin-styled-components/issues/18#issuecomment-271529646)

**Comment by santaclauze** [comment]
Nevermind, Solved @kopax  you were right, due to another error in a css mixin that involved the Navbars. Looking for styled-components 2.0 :)
